### DISCUSSION TITLE
Bug of hydration children for html elements

### DISCUSSION THREAD

**Discussion by Shoko118** [nuxt/nuxt#10232] [Questions (Nuxt 2)] [answered]
### Versions

- nuxt: 3.0.0

## Describe the bug 
Hello Nuxt Team 😄 
This bug I have been facing is that in my projects  there is a lot html div , a , p . .. etc .
But in small project there isn't any issue for that so I was thinking that is it because my project  has a lot html that's why the bug it's so random ??
And also I've been searching a lot of ways and I can't find the answer 😢 
## My Code
```   
<!-- index.vue-->
<template>
      <div>
        ..... alot of div and etc
        ..... components
       </div>
</template>
```
## Bug code 
~~~
instrument.ts:130 [Vue warn]: Hydration children mismatch in <a>: server rendered element contains fewer child nodes than client vdom. 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > key="/" > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="page" mode="out-in" > 
  at <Custom name="custom" > 
  at <AsyncComponentWrapper name="custom" > 
  at <NuxtLayout key=0 name="custom" > 
  at <RouterView> 
  at <NuxtPage> 
  at <App> 
  at <NuxtRoot>

~~~
~~~
[Vue warn]: Hydration children mismatch in <p>: server rendered element contains fewer child nodes than client vdom. 
  at <SecondComponent banner="system" > 
  at <CompleteTech onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > key="/complete-tech" > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="page" mode="out-in" > 
  at <Custom name="custom" > 
  at <AsyncComponentWrapper name="custom" > 
  at <NuxtLayout key=0 name="custom" > 
  at <RouterView> 
  at <NuxtPage> 
  at <App> 
  at <NuxtRoot>
~~~
~~~
[Vue warn]: Hydration children mismatch in <a>: server rendered element contains fewer child nodes than client vdom. 
  at <FirstComponent banner="system" > 
  at <CompleteTech onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > key="/complete-tech" > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="page" mode="out-in" > 
  at <Custom name="custom" > 
  at <AsyncComponentWrapper name="custom" > 
  at <NuxtLayout key=0 name="custom" > 
  at <RouterView> 
  at <NuxtPage> 
  at <App> 
  at <NuxtRoot>
~~~

**Comment by danielroe** [comment, marked as answer]
I suspect you are nesting HTML elements that aren't allowed within their ancestor HTML element.

For example, `<a>` cannot be within another `<a>`. The same goes for `<p>`, which also cannot contain any other block elements, like `<div>`.

**Reply by brianacdev** [reply]
This ended up being my issue. I had a component called `Header.vue` that I was using in my layout and then I had a `header` element inside that component. Turns out a `header` can't be inside a `header`

**Comment by nosizejosh** [comment]
I am having a similar issue with headlessui tabs. 

```
[Vue warn]: Hydration children mismatch in <div>: server rendered element contains more child nodes than client vdom. 
  at <TabList class="inline-flex rounded-md shadow-sm" role="group" > 
  at <Header> 
  at <TabGroup defaultIndex=0 > 
  at <Index> 
  at <AppLayout > 
  at <LayoutLoader key="app-layout" name="app-layout" hasTransition=false > 
  at <FragmentWrapper > 
  at <NuxtLayout name="app-layout" > 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <RouteProvider key="/ford-motor-corp/pipelines" routeProps= {Component: {…}, route: {…}} pageKey="/ford-motor-corp/pipelines"  ... > 
  at <FragmentWrapper > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <App key=2 > 
  at <NuxtRoot>
```

what is strange about my issue is that, though the waring is always present, the styling of the tabs are only affected and messed up when the page is visited from the address bar or on refresh. If the page is visited by navigating from another page, the warning doesn't affect the styling of the tabs and everything is fine. 

I'm on Nuxt 3.2.3 and I have tried to fix this with styling or applying clientOnly around the header that contains the tabs but nothing has worked so far. 

Please help.

**Reply by claide** [reply]
@nosizejosh got the same issue yesterday as well. Did you manage to fix it?

What I tries so far is commenting all custom components inside a page but to no avail.

**Reply by nosizejosh** [reply]
hey @claide no I wasn't able to fix it. for my case I needed a simple tab solution so I rolled out a DIY version which stopped the warnings.

**Reply by cn1001wang** [reply]
when I change page index, I got it.

**Comment by softdevca** [comment]
I have seen this same error when I had child component with a <template> but no contents.

**Reply by fmarcheski** [reply]
This was my issue. I had a child div that was "hidden" for desktop, which was causing hydration issues. Instead of using "hidden" I added a v-if="$device.isMobile" for the div, which solved my problem.

Thanks!

**Comment by fsarter** [comment]
I ran the same issue when using `useAsyncData` to fetch server api data and render them in the component, then i found that i forgot `await` before method call.

**Comment by amirho3ein83** [comment]
<img width="890" height="710" alt="solved" src="https://github.com/user-attachments/assets/4de11ce4-1c80-4973-9bf1-78c25122d7f3" />
I solved my problem by removing the unnecessary "a" tag
